### PR TITLE
fix: Update RedisCluster to use redis-py official implementation

### LINKED ISSUE #2552: Switch from `redis-py-cluster` to `redis-py`
As [documented](https://github.com/Grokzen/redis-py-cluster#redis-py-cluster-eol), `redis-py-cluster` has reached EOL. It is now being merged into [`redis-py`](https://github.com/redis/redis-py#cluster-mode). We previously tried to switch from `redis-py-cluster` to `redis-py` but ran into various issues; see #2328. The upstream [bug](https://github.com/redis/redis-py/issues/2003) has since been fixed, so we can now switch from `redis-py-cluster` to `redis-py`.

### PR DESCRIPTION
**Pull request by kevjumba**
**What this PR does / why we need it**:
Update redis cluster implementation and remove redis cluster dependency. Also update crypto for mac m1 to 3.4.8(this is the newest version that patches some fixes in order to be compatible with openssl 1.1.1(the older version has been giving people with mac m1 trouble in installation).

**Which issue(s) this PR fixes**:

Fixes #2552

### CODE REVIEW

File: sdk/python/requirements/py3.10-ci-requirements.txt [hunk 1]
```diff
@@ -208,22 +204,24 @@ google-auth==1.35.0
     #   google-cloud-storage
 google-auth-httplib2==0.1.0
     # via google-api-python-client
-google-auth-oauthlib==0.5.0
+google-auth-oauthlib==0.5.1
     # via gcsfs
-google-cloud-bigquery==2.34.1
-    # via feast (setup.py)
-google-cloud-bigquery-storage==2.12.0
+google-cloud-bigquery==3.0.1
```
Comment (achals) [inline sdk/python/requirements/py3.10-ci-requirements.txt:209]: We should pin this below 3.0.0. I think the 3+ is causing some issues, cc #2537 or #2532

**Review by achals — APPROVED** [review]
Single change

File: sdk/python/requirements/py3.10-ci-requirements.txt (hunk 1 above)
Comment (kevjumba) [reply sdk/python/requirements/py3.10-ci-requirements.txt:209]: Sure, let me do that right now.

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**